subsys/bluetooth/host/hci_core.c

```diff
@@ -3387,6 +3398,27 @@ static int le_init(void)
 		net_buf_unref(rsp);
 	}
 
+#if defined(CONFIG_BT_BROADCASTER)
+	if (IS_ENABLED(CONFIG_BT_BROADCASTER)) {
+		/* Read LE Max Adv Data Len */
+		err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_READ_MAX_ADV_DATA_LEN, NULL, &rsp);
+		if (err == -EIO) {
+			/* if the command doesn't exist, flag it with 0xffff so
+			 * the host knows that it doesn't know the maximum
+			 * advertising size of the controller
+			 * we can use 0xffff because the maximum length of
+			 * advertising data is 1650
+			 */
+			bt_dev.le.max_adv_data_len = 0xffff;
+		} else if (err != 0) {
```

Comment:
`err < 0` is more common

Reply:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/2520413/6ec35870-ea9a-4f2f-b05e-db3a0290b741)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/2520413/0170c3f5-8ad6-4f8e-a205-8080798dcdc3)

Not in Zephyr :)

Reply:
`if (err)` might even be the most common form in Bluetooth code. We should try to standardize on one style and include it in #55281. I kind of like the `if (err)` form since it's closest to natural language ("if there is an error, then ...") however it's not compliant with MISRA as it treats an integer as a boolean. One risk that both `if (err)` and `if (err != 0)` have is that we have some APIs where a positive value indicates success in the form of some size and a negative value is an error.